sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -189,8 +189,17 @@ case class AdaptiveSparkPlanExec(
           stagesToReplace = result.newStages ++ stagesToReplace
           executionId.foreach(onUpdatePlan(_, result.newStages.map(_.plan)))
 
+          // SPARK-33933: We should materialize BroadcastQueryState first to avoid broadcast timeout
+          // Sort the new stages by class type to make sure BroadcastQueryState precede others
```

Comment:
nit:
```
// SPARK-33933: we should submit tasks of broadcast stages first, to avoid broadcast timeout.
```

Reply:
I think a better solution is to exclude task scheduling time in broadcast timeout, but it's hard to implement and we should have this workaround first.

cc @wangyum @viirya

Reply:
Agreed